src/core/src/pso.rs

```diff
@@ -258,27 +259,36 @@ impl<R: Resources> PixelTargetSet<R> {
             depth: None,
             stencil: None,
             size: (0, 0, 0, texture::AaMode::Single),
+            sized: false,
         }
     }
+
     /// Add a color view to the specified slot
-    pub fn add_color(&mut self, slot: ColorSlot, view: &R::RenderTargetView,
+    pub fn add_color(&mut self,
+                     slot: ColorSlot,
+                     view: &R::RenderTargetView,
                      dim: texture::Dimensions) {
-        use std::cmp::max;
         self.colors[slot as usize] = Some(view.clone());
-        self.size = max(self.size, dim);
+        self.set_size(dim);
     }
+
     /// Add a depth or stencil view to the specified slot
     pub fn add_depth_stencil(&mut self, view: &R::DepthStencilView,
                              has_depth: bool, has_stencil: bool,
                              dim: texture::Dimensions) {
-        use std::cmp::max;
         if has_depth {
             self.depth = Some(view.clone());
         }
         if has_stencil {
             self.stencil = Some(view.clone());
         }
-        self.size = max(self.size, dim);
+        self.set_size(dim);
+    }
+
+    fn set_size(&mut self, dim: texture::Dimensions) {
+        debug_assert!(!self.sized || self.size == dim);
+        self.sized = true;
+        self.size = dim;
```

Comment:
perhaps, use `unwrap_or` at least, may be changed later to some error if we want